client/components/Editor/index.jsx

```diff
@@ -69,6 +111,46 @@ class Editor extends Component {
         this.props.history.push('/');
     }
 
+    async updateScenario(scenario) {
+        scenario = scenario || this.state.scenario;
+
+        if (this.state.saving) {
```

Comment:
To make sure I understand this logic, we don't want to update the scenario if changes are in the process of being sent to the database? If so. what happens to the changes that don't get saved because it was in this process?